src/librustc/ty/context.rs

```diff
@@ -853,6 +853,11 @@ pub struct GlobalCtxt<'tcx> {
 
     pub dep_graph: DepGraph,
 
+    /// This provides access to the incr. comp. on-disk cache for query results.
+    /// Do not access this directly. It is only meant to be used by
+    /// `DepGraph::try_mark_green()` and the query infrastructure in `ty::maps`.
+    pub(crate) on_disk_query_result_cache: maps::OnDiskCache<'tcx>,
```

Comment:
as an aside, it feels to me like "`pub(crate)` visibility with scary comment" is a sign that this code is not "well aligned" in terms of module structure. We should review at some point.